lib/delete-users-using-local-file.rb

```diff
@@ -0,0 +1,109 @@
+require 'zendesk_api'
+require 'rest-client'
+require 'json'
+require 'date'
+
+require_relative 'zendesk-setup.rb'
+
+# Setup vars
+
+diag = "true"
+url = "#{ENV['ZENDESK_URL']}/deleted_users/"
+lastyear = Date.today.next_day - 365
+
+File.open "data/soft_deleted_users.json" do |input_file|
+
+  file_data = JSON.load input_file
+
+  file_data.each do |user|
+
+
+    # Extract some fields
+    updated_at = user["updated_at"]
+    last_login_at = user["last_login_at"]
+    user_id = user["id"]
+    active = user["active"]
+    name = user["name"]
+
+    # Convert strings to dates, if strings are nil then set a false date of 2012 so comparison works.
+    # Booleans returned by API are not true boolean but strings.
+
+    if active == "false"
+      # account is already soft deleted so force hard delete
+      hard_delete = "true"
+    end
+
+    false_date = "2012-01-01"
+
+    if hard_delete != "true"
+      if updated_at.nil?
+        updated_at = false_date
+      end
+
+      # parse the date so we can do comparisons
+
+      updated = Date.parse(updated_at)
```

Comment:
`updated` variable is not used. Can we remove it?

Reply:
Resolved as suggested.